### PR TITLE
feat: switch to `abstract-level`

### LINKED ISSUE #585: switch to `abstract-level` from `abstract-leveldown`
It looks like the level community is deprecating [`abstract-leveldown`](https://github.com/Level/abstract-leveldown) in favor of [`abstract-level`](https://github.com/Level/abstract-level). They published some [nice upgrade guides](https://github.com/Level/community#how-do-i-upgrade-to-abstract-level) that look straight forward enough.

### PR DESCRIPTION
**Pull request by mikaelkaron**
closes #585

### CODE REVIEW

**Comment by mikaelkaron** [comment]
This depends on https://github.com/fergiemcdowall/fergies-inverted-index/pull/44 being merged and released, and after that updating `package.json`.

File: dist/search-index-3.1.3.js.LICENSE.txt [hunk 1]
```diff
@@ -1,10 +1,3 @@
-/*!
- * Determine if an object is a Buffer
- *
- * @author   Feross Aboukhadijeh <https://feross.org>
- * @license  MIT
- */
-
 /*!
```
Comment (fergiemcdowall) [inline dist/search-index-3.1.3.js.LICENSE.txt:1]: Thanks for this @mikaelkaron ! I have no idea how that got in there

File: dist/search-index-esm-3.1.3.js.LICENSE.txt [hunk 2]
```diff
@@ -1,10 +1,3 @@
-/*!
- * Determine if an object is a Buffer
- *
- * @author   Feross Aboukhadijeh <https://feross.org>
- * @license  MIT
- */
-
```
Comment (fergiemcdowall) [inline dist/search-index-esm-3.1.3.js.LICENSE.txt:7]: 👍👍

File: src/read.js [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 
+const { EntryStream } = require('level-read-stream')
+
 module.exports = (ops, cache) => {
   // TODO add aggregation to ALL_DOCUMENTS
   const ALL_DOCUMENTS = limit =>
     new Promise((resolve, reject) => {
       const result = []
-      ops.fii.STORE.createReadStream({
+      new EntryStream(ops.fii.STORE, {
```
Comment (fergiemcdowall) [inline src/read.js:10]: Much cleaner than my old code- nice!

**Review by fergiemcdowall — DISMISSED** [review]
Changes all look good 👍👍

File: test/src/memory-level-test.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
-const memdown = require('memdown')
-const si = require('../../')
+const { MemoryLevel } = require('memory-level')
+const si = require('../../src/node')
```
Comment (fergiemcdowall) [inline test/src/memory-level-test.js:2]: @mikaelkaron I think that this line is breaking the build and that it should be:

```javascript
const si = require('../../')
```

Other that it all looks good (as far as I can see)

**Review by fergiemcdowall — CHANGES_REQUESTED** [review]
What looks like a small mistake in memdown-test.js is breaking the build. Otherwise all good 👍👍

File: test/src/memory-level-test.js (hunk 4 above)
Comment (fergiemcdowall) [reply test/src/memory-level-test.js:2]: (line 2)

Comment (mikaelkaron) [reply test/src/memory-level-test.js]: clumsy me, that was not intended. fix pushed

**Comment by fergiemcdowall** [comment]
@mikaelkaron good idea re types

**Review by fergiemcdowall — APPROVED** [review]

**Comment by mikaelkaron** [comment]
> @mikaelkaron good idea re types

I've prepared [needed changes](https://github.com/mikaelkaron/DefinitelyTyped/tree/search-index) but waiting for you to release before I submit a PR. This assumes you'll release this as `3.2`.

**Comment by fergiemcdowall** [comment]
Great! `3.2.0` is now out on npm

**Pull request merged**